### PR TITLE
Allow Events to be placed on Water and fix invalid passability for them

### PR DESCRIPTION
**Pull request by ihhub**
relates to #6845

During the gameplay Events are not considered as action objects and they are excluded from overall pathfinding calculations. However, outside the gameplay it is an action object and should be treated properly.

### CODE REVIEW

**Review by Districh-ru — APPROVED** [review]
I tested a couple of maps and did not notice passability issues.
👍

**Comment by ihhub** [comment]
Hi @Districh-ru , I added more changes in this pull request based on @Branikolog feedback.

@oleg-derevenetz , could you please take a look to verify that I am not breaking some critical code?

File: src/fheroes2/maps/mp2.h [hunk 1]
```diff
@@ -499,10 +499,12 @@ namespace MP2
...
     // TODO: make a separate function to determine whether the object is an action object depending on its location and not
     // TODO: on where it is accessed from.
-    bool isActionObject( const MapObjectType objectType, const bool accessedFromWater );
+    bool isInGameActionObject( const MapObjectType objectType, const bool accessedFromWater );
 
     // The method checks if the object is an action object regardless of where it is accessed from.
-    bool isActionObject( const MapObjectType objectType );
+    bool isInGameActionObject( const MapObjectType objectType );
+
+    bool isOffGameActionObject( const MapObjectType objectType );
```
Comment (oleg-derevenetz) [inline src/fheroes2/maps/mp2.h:507]: Hi @ihhub I'd suggest to add a comment for this method describing what exactly it checks and what is its difference from the `isInGameActionObject()`.

Comment (ihhub) [reply src/fheroes2/maps/mp2.h:507]: Sure!

**Review by Branikolog — APPROVED** [review]

**Review by Districh-ru — APPROVED** [review]

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**